test/models/job_execution_test.rb

```diff
@@ -424,16 +422,22 @@ def with_hidden_errors
 
   describe "#pid" do
     it "returns current pid" do
+      job.command = 'sleep 0.5'
       execution = JobExecution.new('master', job)
-      execution.executor.stubs(pid: 2)
-      execution.pid.must_equal 2
+      JobExecution.start_job(execution)
+      sleep 0.5
```

Comment:
not very likely since close etc takes 0.1s or so, but having 0.4 would make things clearer ... I'll update